pkg/service/record/record.go

```diff
@@ -66,13 +67,18 @@ func (r *Recorder) Start(ctx context.Context) error {
 	var appID uint64
 	var newTestSetID string
 	var testCount = 0
+	var mutex = &sync.Mutex{}
+	var waitGroup = &sync.WaitGroup{}
 	var mockCountMap = make(map[string]int)
 
 	// defering the stop function to stop keploy in case of any error in record or in case of context cancellation
 	defer func() {
 		select {
 		case <-ctx.Done():
+			waitGroup.Wait()
+			mutex.Lock()
 			r.telemetry.RecordedTestSuite(newTestSetID, testCount, mockCountMap)
```

Comment:
I might be missing something but not really sure if this defer could be called concurrently 🤔

Reply:
aah, so its the testCount probably..

Reply:
I think in the defer function if we wait for the other goroutines to complete/cancel - this race condition can be better resolved. Maybe we can move the `runAppErrGrp.Wait()` in the defer to the top. 

@charankamarapu @gouravkrosx what do you think?

Reply:
@dxtym , we can move the `r.telemetry.RecordedTestSuite(newTestSetID, testCount, mockCountMap)` to the bottom of the defer, this way we are waiting for all the go routines to complete their execution.

Reply:
Thank you for the comments @slayerjain @gouravkrosx! I moved the `r.telemetry.RecordedTestSuite(newTestSetID, testCount, mockCountMap)` to the bottom and removed mutex and waitGroups. Did I get your feedback right?

Reply:
@dxtym yes, I think you've made the changes according to the feedback. 

It seems like `ctx` cancellation is not working correctly based on the this pipeline - https://github.com/keploy/keploy/actions/runs/9034964803/job/24829105617?pr=1867


It seems to be stuck on the last line: 

<img width="1127" alt="Screenshot 2024-05-10 at 9 55 22 PM" src="https://github.com/keploy/keploy/assets/12831254/2e2f49e9-42a2-43f6-b115-4be02abc383e">

Reply:
![image](https://github.com/keploy/keploy/assets/100064552/f11e2035-fcf8-4993-bc99-46045d54075c)
I checked one of the previous PRs that passed all the tests. It had the same line. Could it be something else?

Reply:
What if I undo the changes to `ctx` cancelling part by not moving down? I did the previous solution with `mutex` while removing  the `waitGroup`, and no race condition occurred.  Perhaps, we could go with that )

Reply:
I tried removing `waitGroup` from my previous solution, and the binary worked the same way. Perhaps we could try that )